src/ai/backend/gateway/domainconfig.py

```diff
@@ -0,0 +1,194 @@
+    if not request['is_superadmin'] and not request['is_admin']:
+        raise GenericForbidden('Only superadmin or domain admin can create dotfiles')
+    elif request['is_admin'] and request['user']['domain_name'] != params['domain']:
+        raise GenericForbidden('Domain-admins cannot create dotfiles of other domains')
+
+    dbpool = request.app['dbpool']
+    async with dbpool.acquire() as conn, conn.begin():
+        dotfiles, leftover_space = await query_domain_dotfiles(conn, params['domain'])
+        if dotfiles is None:
+            raise DomainNotFound('Input domain is not found')
+        if leftover_space == 0:
+            raise DotfileCreationFailed('No leftover space for dotfile storage')
+        if len(dotfiles) == 100:
+            raise DotfileCreationFailed('Dotfile creation limit reached')
+        if not verify_dotfile_name(params['path']):
+            raise InvalidAPIParameters('dotfile path is reserved for internal operations.')
+
+        duplicate = [x for x in dotfiles if x['path'] == params['path']]
+        if len(duplicate) > 0:
+            raise DotfileAlreadyExists
+        new_dotfiles = list(dotfiles)
+        new_dotfiles.append({'path': params['path'], 'perm': params['permission'],
+                             'data': params['data']})
+        dotfile_packed = msgpack.packb(new_dotfiles)
+        if len(dotfile_packed) > MAXIMUM_DOTFILE_SIZE:
+            raise DotfileCreationFailed('No leftover space for dotfile storage')
+
+        query = (domains.update()
+                        .values(dotfiles=dotfile_packed)
+                        .where(domains.c.name == params['domain']))
+        await conn.execute(query)
+    return web.json_response({})
+
+
+@auth_required
+@server_status_required(READ_ALLOWED)
+@check_api_params(t.Dict({
+    t.Key('domain'): t.String,
+    t.Key('path', default=None): t.Null | t.String,
+}))
+async def list_or_get(request: web.Request, params: Any) -> web.Response:
+    resp = []
+    dbpool = request.app['dbpool']
+
+    log.info('LIST_OR_GET DOTFILE (domain: {0})', params['domain'])
+    async with dbpool.acquire() as conn:
+        if params['path']:
+            dotfiles, _ = await query_domain_dotfiles(conn, params['domain'])
+            if dotfiles is None:
+                raise DomainNotFound
+            for dotfile in dotfiles:
+                if dotfile['path'] == params['path']:
+                    return web.json_response(dotfile)
+            raise DotfileNotFound
+        else:
+            dotfiles, _ = await query_domain_dotfiles(conn, params['domain'])
+            if dotfiles is None:
+                raise DomainNotFound
+            for entry in dotfiles:
+                resp.append({
+                    'path': entry['path'],
+                    'permission': entry['perm'],
+                    'data': entry['data']
+                })
+            return web.json_response(resp)
+
+
+@server_status_required(READ_ALLOWED)
+@auth_required
```

Comment:
Domain dotfile should not be updated by normal uses. There should be an authorization check like creation API.